### ISSUE TITLE
Typo in warning message can cause Jupyter start to fail

### ISSUE THREAD

**Issue by malcolmbovey** [jupyter-server/jupyter_server#422]
There is a typo in serverapp.py, line 2020 in jupyter_server v.1.4.0 which cause startup to fail when there is no browser

File "/opt/conda/lib/python3.8/site-packages/jupyter_server/serverapp.py", line 2020, in launch_browser
jupyter_1 | self.log.warning(_('No web browser found: %s.') % e)
UnboundLocalError: local variable _ referenced before assignment

**Comment by ChristophSchranz** [comment]
Same here:

```
gpu-jupyter_1  | [I 2021-02-22 13:38:20.750 ServerApp] Serving notebooks from local directory: /home/jovyan
gpu-jupyter_1  | [I 2021-02-22 13:38:20.750 ServerApp] Jupyter Server 1.4.0 is running at:
gpu-jupyter_1  | [I 2021-02-22 13:38:20.750 ServerApp] http://af6150fe4d6f:8888/lab
gpu-jupyter_1  | [I 2021-02-22 13:38:20.750 ServerApp]  or http://127.0.0.1:8888/lab
gpu-jupyter_1  | [I 2021-02-22 13:38:20.750 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
gpu-jupyter_1  | Traceback (most recent call last):
gpu-jupyter_1  |   File "/opt/conda/lib/python3.8/site-packages/jupyter_server/serverapp.py", line 2018, in launch_browser
gpu-jupyter_1  |     browser = webbrowser.get(self.browser or None)
gpu-jupyter_1  |   File "/opt/conda/lib/python3.8/webbrowser.py", line 65, in get
gpu-jupyter_1  |     raise Error("could not locate runnable browser")
gpu-jupyter_1  | webbrowser.Error: could not locate runnable browser
gpu-jupyter_1  | 
gpu-jupyter_1  | During handling of the above exception, another exception occurred:
gpu-jupyter_1  | 
gpu-jupyter_1  | Traceback (most recent call last):
gpu-jupyter_1  |   File "/opt/conda/bin/jupyter-lab", line 10, in <module>
gpu-jupyter_1  |     sys.exit(main())
gpu-jupyter_1  |   File "/opt/conda/lib/python3.8/site-packages/jupyter_server/extension/application.py", line 520, in launch_instance
gpu-jupyter_1  |     serverapp.start()
gpu-jupyter_1  |   File "/opt/conda/lib/python3.8/site-packages/jupyter_server/serverapp.py", line 2100, in start
gpu-jupyter_1  |     self.start_app()
gpu-jupyter_1  |   File "/opt/conda/lib/python3.8/site-packages/jupyter_server/serverapp.py", line 2059, in start_app
gpu-jupyter_1  |     self.launch_browser()
gpu-jupyter_1  |   File "/opt/conda/lib/python3.8/site-packages/jupyter_server/serverapp.py", line 2020, in launch_browser
gpu-jupyter_1  |     self.log.warning(_('No web browser found: %s.') % e)
gpu-jupyter_1  | UnboundLocalError: local variable '_' referenced before assignment
```

**Comment by malcolmbovey** [comment]
To clarify - I am also running Jupyter in docker when I get this error.

**Comment by kevin-bates** [comment]
Fixed in #421

**Comment by Zsailer** [comment]
Thanks, @malcolmbovey. We will cut a patch release today with #421.

**Comment by Zsailer** [comment]
Fixed in [1.4.1](https://pypi.org/project/jupyter-server/1.4.1/).
